### DISCUSSION TITLE
JSONB query match from specific value in array of arrays

### DISCUSSION THREAD

**Discussion by NanezX** [supabase/supabase#10597] [Questions]
Hello, hope everyone is good.

I'm little new on supabase and I'm wondering about how to make an query with a specific patten from a JSONB in our table. The json is in the column `metadata` and it is more complex, but I can query almost all the info. Have the following form, like:

```
{
  id:  1234
  addresses: [
    {
      chainId: 1,
      knownAddresses: ['0x1234', '0x5678']
    },
    {
      chainId: 2,
      knownAddresses: ['0x4321', '0x8765']
    },
  ]
},
{
  id:  4321
  addresses: [
    {
      chainId: 1,
      knownAddresses: ['0xabcd', '0xefgh']
    },
    {
      chainId: 2,
      knownAddresses: ['0xdcba', '0xhgfe']
    },
  ]
}
```

What I want to query is: Check if a string like `'0x8765'` exist in `addresses` field of my JSON, and return only that with the correct `chainId`. I tried a lot of combinations without success, but something like:

```
const query = await supabaseClient
	.from('myTable')
	.select('metadata->addresses, id')
	.contains('metadata->addresses', JSON.stringify([{ knownAddresses: ['0x8765'] }]));
```

But this does not match nothing and I spent the whole day on this. Is there any way to reach this?

Any idea or guide to find out will be awesome, thanks for your time.

**Comment by GaryAustin1** [comment]
Your comments don't seem to match your column data.  Are you trying to search knownAddresses array for a value?  You are searching addresses....

contains() must always compare an array column (jsonb field) with a fixed array.  So stringify won't work.   Plus the array in your case is knownAddresses (you don't have a jsonb array with keys of knownAddresses in it).

Maybe something  like .contains('metadata->addresses->knownAddresses', '{0x8765}')  
BUT I'm not sure on the array format '{0x9765}'.... '{}' is a postgres array in a string, but I don't know if something needs to go around 0x9765 or if that is even a string.   Like '{"0x8765"}'...

**Reply by NanezX** [reply]
Yes. I'm searching for a value on knownAddresses but yeah -  I explained myself wrongly.

I wanted get the match for a value on knownAddresses , but idk if `.contains('metadata->addresses->knownAddresses', '{0x8765}')` will work since `addresses` it's an arrar. I understand something like `.contains('metadata->addresses->index->knownAddresses', '{0x8765}')`  but can I match against all those arrays?

**Reply by GaryAustin1** [reply]
Ah yeah, the searching the array is not the issue, the fact you have several objects in the column is. I’m not good enough with json and Postgres for that.  I do think it is not going to be something you can do with the api though.  If it can be done with SQL you could use an rpc call.
